tests/Doctrine/Tests/Common/Annotations/AnnotationRegistryTest.php

```diff
@@ -168,4 +168,25 @@ public function testResetClearsRegisteredAutoloaderFailures() : void
 
         self::assertSame(2, $failures);
     }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testRegisterLoaderIfNotExistsOnlyRegisteresSameLoaderOnce() : void
+    {
+        $className = 'autoloadedClass' . random_int(10, 100000);
+        AnnotationRegistry::reset();
+        $i = 0;
+        $autoLoader = function () use (&$i) : bool {
+            $i += 1;
+            return false;
+        };
```

Comment:
Not a big deal but I'd prefer to use PHPUnit mocks where possible instead of adding logic to the test. The mocking framework is tested by its own tests but this logic isn't.

The above could be implemented as:
```php
$autoLoader = self::createPartialMock(stdClass::class, ['__invoke']);
$autoLoader->expects($this->once())->method('__invoke');
```

Reply:
well, the fact that the mocking framework is tested does not mean that your own usage performs the right logic with it. And so this offers no guarantee that the test being written tests things properly either.

Reply:
I have no problem changing this, I copied it from other tests in this file. Shouldn't it be the same for consistency?

Reply:
> well, the fact that the mocking framework is tested does not mean that your own usage performs the right logic with it.

True.

> And so this offers no guarantee that the test being written tests things properly either.

True.

It just doesn't look right to implement mocking a callable and counting calls while this is already implemented in a tool which is designed for exactly that purpose.